internal/pkg/service/common/task/result.go

```diff
@@ -3,34 +3,20 @@ package task
 import (
 	"strings"
 
-	"github.com/keboola/keboola-as-code/internal/pkg/telemetry"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 )
 
 type Result struct {
-	result    string
-	error     error
-	errorType string
-	// applicationError flag is true if the error is an unexpected error that should not occur during normal operation
-	applicationError bool
-	outputs          map[string]any
-}
-
-// UserError marks the wrapped error as expected, so it will not be taken as an error in the metrics.
-// UserError is an error that could occur during normal operation.
-type UserError struct {
-	error
-}
-
-func (e UserError) Unwrap() error {
-	return e.error
+	Result  string
+	Error   error
+	Outputs Outputs
 }
```

Comment:
V tomto commite upravujem `task.Result` strukturu:
- Zmenil som ju na "hlupu" value strukturu.
- Private fields som zmenil na Public.
- Odstranil som zbytocnu logiku (presunul inde).

------------

`taks.Result` je vysledkom tasku:
- Ma nastaveny `Result` string ak je task OK.
- Ma nastaveny `Error` ak nastala chyba.
- V oboch pripadoch moze vratit nejake `Outputs`, napr. `exportID`, ked task vytvaral export.
- Kedze su to ^^^ 3 rozne casti, tak to nevracia ako samostatne parametre, ale je to obalene do tejto struktury.

-----

Preco?

Potrebujem dosiahnut nieco take:
![image](https://github.com/keboola/keboola-as-code/assets/19371734/4165f5e4-92b7-4766-846b-d161760e369e)

Tj. na zaver tasku potrebujem pristup ku pripadnej chybe a musim mat moznost ju upravit.
To sa da dosiahnut rozne (napr. boli by tam 3 parametre `outputs Outputs result string, err error`),
ja som vybral tento ^^^ sposob, kedze `outputs` sa nepouzivaju vzdy, a takyto zapis je kratky a jednoduchy.